app/src/main/java/space/taran/arknavigator/mvp/presenter/ResourcesPresenter.kt

```diff
@@ -113,16 +119,74 @@ class ResourcesPresenter(
             tagsSelectorPresenter.calculateTagsAndSelection()
 
             val path = (rootAndFav.fav ?: rootAndFav.root)
-            val title = if (path != null) "$path, " else ""
+            val title = if (path != null) "${path.last()}, " else ""
 
             viewState.setToolbarTitle("$title${roots.size} of roots chosen")
             viewState.setProgressVisibility(false)
         }
     }
 
-    override fun onDestroy() {
-        Log.d(RESOURCES_SCREEN, "destroying ResourcesPresenter")
-        super.onDestroy()
+    fun onMoveSelectedResourcesClicked(
+        directoryToMove: Path
+    ) = presenterScope.launch(Dispatchers.IO) {
+        withContext(Dispatchers.Main) {
+            viewState.setProgressVisibility(true, "Moving")
+        }
+        val resourcesToMove = gridPresenter.resources.filter { it.isSelected }
+        val results = resourcesToMove.map { item ->
+            async {
+                val path = index.getPath(item.meta.id)
+                val newPath = directoryToMove.resolve(path.name)
+                path.copyTo(newPath)
+                if (path != newPath)
+                    path.deleteIfExists()
+            }
+        }
+        results.forEach { it.await() }
+        index.reindex()
```

Comment:
Can we optimize this, telling into index what was moved and updating only affected paths? There should be no `ResourceId` re-calculation, in fact.

Reply:
Id calculation does not occur. We only check files for existence and modification date.

Reply:
However, Id recalculation can happen when moving resources to another folder in the same root. It can be fixed